test/images/agnhost/serve-hostname/serve_hostname.go

```diff
@@ -125,6 +124,4 @@ func main(cmd *cobra.Command, args []string) {
 	signal.Notify(signals, syscall.SIGTERM)
 	sig := <-signals
 	log.Printf("Shutting down after receiving signal: %s.\n", sig)
-	log.Printf("Awaiting pod deletion.\n")
-	time.Sleep(60 * time.Second)
```

Comment:
wow this was controversial https://github.com/kubernetes/kubernetes/pull/54327#discussion_r246415426
let me summon people that was in that discussion
@thockin @BenTheElder 
maybe this sleep is simulating something?

Reply:
I think this was simulating graceful termination in 2 ways:

1) If you use this image behind an LB (including a Service VIP) you can get errors as the LB backends are stale WRT pod terminations.  I don't know if that would matter for tests.

2) If we take this out, any testing that uses this image to prove that the grace-period works will be wrong.  I don't know if such a test exists.  E.g. delete pod, send SIGTERM, pod is still running, 30s grace (or less), kubelet sends SIGKILL, pod is really dead.

I honestly do not have context on it any more, but there's no such thing as a "needless sleep" at this stage in the game :)

If we want to get rid of it in some paths we should probably add (and use) a flag to set it, e.g. `--grace=<duration>`.

Reply:
Isn't it strange to test graceful termination with serve-hostname command, maybe adding a separate command for this purpose would be better?

Reply:
this is not about correctness, the problems is that you can remove it and break some test that depends on it for historical reasons, it seems we have two options
1. verify that we don't break anything removing the sleep (hard to know honestly in a code that grew organically, the presubmits jobs doesn't exercise all the tests)
2. we add a flag as Tim suggests, and we modify all the tests that use the image to use the flag

Reply:
I think this whole conversation IS about correctness, we're trying to make it correct without breaking anything
And setting a flag looks like a good option

Reply:
then I didn't understand you sorry, seems that we all agree on the flag XD